main/src/main/java/cgeo/geocaching/utils/formulas/Value.java

```diff
@@ -62,7 +63,8 @@ public String getAsString() {
             if (raw == null) {
                 asString = "";
             } else if (raw instanceof Number && !(raw instanceof Integer) && !(raw instanceof BigInteger)) {
-                asString = DOUBLE_TO_STRING_FORMAT.format(((Number) raw).doubleValue());
+                final BigDecimal bigDecimalValue = new BigDecimal(((Number) raw).doubleValue());
```

Comment:
It seems odd to create a BigDecimal from a double and expect a more precise outcome from it. (I guess this is also what the codacy warning is about). 

I don't doubt the need of using BigDecimal instead of double, but I feel this should be done from a more fundamental base (using BigDecimal during checksum calculation)

Reply:
> It seems odd to create a BigDecimal from a double and expect a more precise outcome from it. (I guess this is also what the codacy warning is about). 
> 

But that is indeed the case for the transformation to a string value, cause the double creates a scientific string with exponential form, whereas the BigDecimal creates the "long" form

Reply:
> I don't doubt the need of using BigDecimal instead of double, but I feel this should be done from a more fundamental base (using BigDecimal during checksum calculation)


But it's not only for checksum...

Reply:
> > It seems odd to create a BigDecimal from a double and expect a more precise outcome from it. (I guess this is also what the codacy warning is about). 
> > 
> 
> But that is indeed the case for the transformation to a string value, cause the double creates a scientific string with exponential form, whereas the BigDecimal creates the "long" form
> 
> 

I will enhance the test case to show this behavior